lib/MemoryHandler.js

```diff
@@ -59,17 +59,29 @@ MemoryStore.prototype.find = function(request, callback) {
   }
 
   // Return the requested resource
-  return callback(null, theResource);
+  return callback(null, this._clone(theResource));
 };
 
 /**
-  Create (store) a new resource give a resource type and an object.
+  Create (store) a new resource given a resource type and an object.
  */
 MemoryStore.prototype.create = function(request, newResource, callback) {
+  // Check to see if the ID already exists
+  var matches = resources[request.params.type].filter(function(resource) {
+    return resource.id === newResource.id;
+  }).length;
```

Comment:
Now **here** would be a totally legitimate, functional and more efficient (than `.filter`) use of `Array.prototype.some`.